pkg/progressreader/progressreader.go

```diff
@@ -0,0 +1,69 @@
+package progressreader
+
+import (
+	"io"
+)
+
+type StreamFormatter interface {
+	FormatProg(string, string, interface{}) []byte
+	FormatStatus(string, string, ...interface{}) []byte
+	FormatError(error) []byte
+}
+
+type PR_JSONProgress interface {
+	GetCurrent() int
+	GetTotal() int
+}
+
+type JSONProg struct {
```

Comment:
Just remove `JSONProgress` from utils and use strong types instead of `interface{}`